src/client/java/net/neoforged/neoforge/client/neo3d/GpuDeviceProperties.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.neo3d;
+
+import com.mojang.blaze3d.buffers.GpuBuffer;
+import com.mojang.blaze3d.platform.DepthTestFunction;
+import com.mojang.blaze3d.platform.DestFactor;
+import com.mojang.blaze3d.platform.SourceFactor;
+import com.mojang.blaze3d.shaders.ShaderType;
+import com.mojang.blaze3d.shaders.UniformType;
+import com.mojang.blaze3d.textures.AddressMode;
+import com.mojang.blaze3d.textures.GpuTexture;
+import com.mojang.blaze3d.textures.TextureFormat;
+import it.unimi.dsi.fastutil.objects.Object2ReferenceOpenHashMap;
+import it.unimi.dsi.fastutil.objects.ReferenceArrayList;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.Set;
+
+public class GpuDeviceProperties {
+    public final String backendName = defaultVal("Default");
```

Comment:
What is the point of all of this?

Reply:
To report in a uniform way what backend is being used, what API its using, and what version of that API its using without being reliant on building against that backend to do an instanceof check. This is also passed in the configure event which is fired before the `GpuDevice` is available to mods to do a different check against.
`GpuDeviceProperties` itself could be extended to get around that last limitation, but I don't feel that that is a better solution than this.